Description: Improve connection stability
 Without this, ROCK 5B will drop below 2.35 GBps during iperf3 session.
 With this patch, the throughput is much more consistent.
Author: Stephen Chen <stephen@radxa.com>
Forwarded: no
---
diff --git a/src/Makefile b/src/Makefile
index 10a002a..1ec4f7b 100755
--- a/src/Makefile
+++ b/src/Makefile
@@ -38,13 +38,13 @@ CONFIG_DOWN_SPEED_100 = n
 CONFIG_ASPM = y
 ENABLE_S5WOL = y
 ENABLE_S5_KEEP_CURR_MAC = n
-ENABLE_EEE = y
+ENABLE_EEE = n
 ENABLE_S0_MAGIC_PACKET = n
 ENABLE_TX_NO_CLOSE = y
-ENABLE_MULTIPLE_TX_QUEUE = n
+ENABLE_MULTIPLE_TX_QUEUE = y
 ENABLE_PTP_SUPPORT = n
 ENABLE_PTP_MASTER_MODE = n
-ENABLE_RSS_SUPPORT = n
+ENABLE_RSS_SUPPORT = y
 ENABLE_LIB_SUPPORT = n
 ENABLE_USE_FIRMWARE_FILE = n
 DISABLE_PM_SUPPORT = n
diff --git a/src/r8125_n.c b/src/r8125_n.c
index 6ca7201..919ff87 100755
--- a/src/r8125_n.c
+++ b/src/r8125_n.c
@@ -13880,11 +13880,11 @@ rtl8125_init_one(struct pci_dev *pdev,
 #if LINUX_VERSION_CODE < KERNEL_VERSION(3,0,0)
                 tp->cp_cmd |= RxChkSum;
 #else
-                dev->features |= NETIF_F_RXCSUM;
                 dev->hw_features = NETIF_F_SG | NETIF_F_IP_CSUM | NETIF_F_TSO |
                                    NETIF_F_RXCSUM | NETIF_F_HW_VLAN_TX | NETIF_F_HW_VLAN_RX;
                 dev->vlan_features = NETIF_F_SG | NETIF_F_IP_CSUM | NETIF_F_TSO |
                                      NETIF_F_HIGHDMA;
+                dev->features |=  dev->hw_features;
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3,15,0)
                 dev->priv_flags |= IFF_LIVE_ADDR_CHANGE;
 #endif //LINUX_VERSION_CODE >= KERNEL_VERSION(3,15,0)
